taf/testlib/virtual_env.py

```diff
@@ -376,75 +363,55 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
         self._create_subnet(public_network['id'], **subnet_kwargs)
 
         _net_cfg.public_network_id = public_network['id']
+        return _net_cfg.public_network_id
 
-        if create_external_router:
-            router_kwargs = {
-                'routers_client': routers_client,
-                'network_id': _net_cfg.public_network_id,
-                'tenant_id': tenant_id,
-                'enable_snat': True,
-            }
-            public_router = self.create_router(**router_kwargs)
-            assert public_router
-            _net_cfg.public_router_id = public_router['id']
-
-        return  _net_cfg.public_network_id and (not create_external_router or _net_cfg.public_router_id)
-
-    def _get_external_elements(self, routers_client=None):
-        if not routers_client:
-            routers_client = self.handle.os_adm.routers_client
-
+    def _get_external_elements(self):
+        routers_client = self.handle.os_adm.routers_client
         net_filter = {'router:external': True}
-        nets = [net['id'] for net in self.handle._list_networks(**net_filter)]
-        router_2_net_map = {}
+        net_2_router_map = {net['id']: [] for net in self.handle._list_networks(**net_filter)}
         routers_resp = routers_client.list_routers()
         for router in routers_resp['routers']:
             ext_gw_info = router.get('external_gateway_info')
             if ext_gw_info:
                 net_id = ext_gw_info.get('network_id')
-                if net_id and net_id in nets:
-                    router_2_net_map[router['id']] = net_id
+                if net_id and net_id in net_2_router_map:
+                    net_2_router_map[net_id].append(router['id'])
 
-        return router_2_net_map
+        return net_2_router_map
 
-    def _delete_external_elements(self, routers_client=None, networks_client=None,
-                                  ports_client=None):
+    def _delete_external_elements(self):
```

Comment:
Why to delete external elements at all ?  If they were created by taf then cleanup was done. If they were created externally e.g. via magnum deployment then we should not destroy it.
During automated test runs correct deployment is done before the test.